packages/nuxt/src/app/composables/manifest.ts

```diff
@@ -52,7 +54,10 @@ export function getAppManifest (): Promise<NuxtAppManifest> {
 }
 
 /** @since 3.7.4 */
-export async function getRouteRules (url: string) {
+export async function getRouteRules (url: string): Promise<Record<string, any>>
+export async function getRouteRules (event: H3Event): Promise<NitroRouteRules>
+export async function getRouteRules (url: string | H3Event) {
```

Comment:
I wish we never overlapped but perhaps would be safer to deprecate `string` version and support `{ path: string }` interface?

Reply:
the key thing that's missing from the nitro implementation is the ability to get the route rules for an arbitrary path...

might it be possible for nitro to handle a path rather than just the current event?

Reply:
Possible, yes but I'm afraid to do this because it imposes a new limit that route rules cannot be aware of event context anymore. In meanwhile, Nuxt impl can be API-compatible with event interface.